linked-lists/c-or-cpp/merge.cpp

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+
+template <typename T>
+class LinkedList{
+	public:
+		int size = 0;
+		void pushfront(const T& data);
+		T& front();	
+		void popfront();
+		void display();
+		void merge(const LinkedList<T>& list2) ;
+	private:
+		struct node{
+			T data;
+			node* next;
+			node(const T& data): data(data), next(nullptr){}
+		};
+		node* _head = nullptr;
+};
+
+template <typename T>
+T& LinkedList<T>::front(){
+	//returns the first node data of the linked list
+
+	return _head->data;
+}
+
+template <typename T>
+void LinkedList<T>::popfront(){
+	//sets the head pointer to nullptr and the next node as the new head
+	
+	//if there is no node then do nothing
+	if(_head == nullptr) return;
+	if(_head->next == nullptr){
+		_head = nullptr;
+		#delete _head;
```

Comment:
invalid preprocessor directive `#delete`
same on line 42.

also if this is `delete` operator then it should be before assigning head to `nullptr`, as now it is just deleting a null pointer and previous statement is causing a memory leak.

Reply:
got it, #delete is a mistake on my part, i confused it with python commenting.